Neo4j.Driver/Neo4j.Driver/TransactionConfig.cs

```diff
@@ -16,63 +16,61 @@
-/// session.BeginTransaction(b=>{});
-/// </code>
 /// </summary>
 public sealed class TransactionConfig
 {
     internal static readonly TransactionConfig Default = new();
-    private IDictionary<string, object> _metadata;
-    private TimeSpan? _timeout;
-
-    internal TransactionConfig()
-    {
-        _timeout = null;
-        _metadata = PackStream.EmptyDictionary;
-    }
-
-    internal TransactionConfig(IDictionary<string, object> metadata, TimeSpan? timeout)
-    {
-        _metadata = metadata;
-        _timeout = timeout;
-    }
+    internal IDictionary<string, object> _metadata = new Dictionary<string, object>();
+    internal TimeSpan? _timeout = null;
 
     /// <summary>
-    /// Transaction timeout. Transactions that execute longer than the configured timeout will be terminated
-    /// by the database. This functionality allows user code to limit query/transaction execution time. The specified timeout
-    /// overrides the default timeout configured in the database using the <code>db.transaction.timeout</code> setting 
-    /// (<code>dbms.transaction.timeout</code> before Neo4j 5.0). Values higher than <code>db.transaction.timeout</code> will be
-    /// ignored and will fall back to the default for server versions between 4.2 and 5.2 (inclusive). Leave this field unmodified or set it
-    /// to <code>null</code> to use the default timeout configured on the server. A timeout of zero will make the transaction
-    /// execute indefinitely.
+    /// Transaction timeout. Transactions that execute longer than the configured timeout will be terminated by the
+    /// database. This functionality allows user code to limit query/transaction execution time. The specified timeout
+    /// overrides the default timeout configured in the database using the <code>db.transaction.timeout</code> setting (
+    /// <code>dbms.transaction.timeout</code> before Neo4j 5.0). Values higher than <code>db.transaction.timeout</code> will be
+    /// ignored and will fall back to the default for server versions between 4.2 and 5.2 (inclusive). Leave this field
+    /// unmodified or set it to <code>null</code> to use the default timeout configured on the server. A timeout of zero will
+    /// make the transaction execute indefinitely.
     /// </summary>
+    /// <remarks>All positive non-whole millisecond values will be rounded to the next whole millisecond.</remarks>
+    /// <remarks><see cref="TimeSpan.MaxValue"/> ticks will be ignored.</remarks>
     /// <exception cref="ArgumentOutOfRangeException">
-    /// If the value given to transaction timeout in milliseconds is less than zero.
+    /// If the value given to transaction timeout in milliseconds is less than
+    /// zero.
     /// </exception>
     public TimeSpan? Timeout
     {
         get => _timeout;
-        internal set
+        init
         {
-            if ((value ?? TimeSpan.Zero) < TimeSpan.Zero)
+            if (!value.HasValue || value.Value == TimeSpan.MaxValue)
```

Comment:
Is this correct? If value doesn't have a value you are still using it to set _timeout...?